src/librustc_trans/debuginfo/mod.rs

```diff
@@ -390,7 +397,14 @@ pub fn create_function_debug_context<'a, 'tcx>(cx: &CodegenCx<'a, 'tcx>,
 
         // Again, only create type information if full debuginfo is enabled
         let template_params: Vec<_> = if cx.sess().opts.debuginfo == FullDebugInfo {
-            let names = get_type_parameter_names(cx, generics);
+            let names = get_parameter_names(cx, generics);
+            let names = names.iter().flat_map(|(kind, param)| {
+                if kind == &Kind::Type {
+                    Some(param)
+                } else {
+                    None
+                }
+            });
             substs.types().zip(names).map(|(ty, name)| {
```

Comment:
I think my suggestion was to zip `substs` with `names` to have `(Kind, InternedString)` elements (where `Kind` is the one in `ty::subst` - you wouldn't need your own enum here) and only check `k.as_type()`.